example/lib/mobile_scanner_overlay.dart

```diff
@@ -1,228 +1,192 @@
-// //       ),
-// //       body: Center(
-// //         child: Column(
-// //           crossAxisAlignment: CrossAxisAlignment.center,
-// //           mainAxisAlignment: MainAxisAlignment.center,
-// //           children: <Widget>[
-// //             Expanded(
-// //                 child: camState
-// //                     ? Center(
-// //                   child: SizedBox(
-// //                     width: 300.0,
-// //                     height: 600.0,
-// //                     child: MobileScanner(
-// //                       onError: (context, error) => Text(
-// //                         error.toString(),
-// //                         style: TextStyle(color: Colors.red),
-// //                       ),
-// //                       qrCodeCallback: (code) {
-// //                         setState(() {
-// //                           qr = code;
-// //                         });
-// //                       },
-// //                       child: Container(
-// //                         decoration: BoxDecoration(
-// //                           color: Colors.transparent,
-// //                           border: Border.all(
-// //                               color: Colors.orange,
-// //                               width: 10.0,
-// //                               style: BorderStyle.solid),
-// //                         ),
-// //                       ),
-// //                     ),
-// //                   ),
-// //                 )
-// //                     : Center(child: Text("Camera inactive"))),
-// //             Text("QRCODE: $qr"),
-// //           ],
-// //         ),
-// //       ),
-// //       floatingActionButton: FloatingActionButton(
-// //           child: Text(
-// //             "press me",
-// //             textAlign: TextAlign.center,
-// //           ),
-// //           onPressed: () {
-// //             setState(() {
-// //               camState = !camState;
-// //             });
-// //           }),
-// //     );
-// //   }
-// // }
+import 'package:flutter/material.dart';
+import 'package:mobile_scanner/mobile_scanner.dart';
+import 'package:mobile_scanner_example/scanner_error_widget.dart';
+
+class BarcodeScannerWithOverlay extends StatefulWidget {
+  @override
+  _BarcodeScannerWithOverlayState createState() =>
+      _BarcodeScannerWithOverlayState();
+}
+
+class _BarcodeScannerWithOverlayState extends State<BarcodeScannerWithOverlay> {
+  String overlayText = "Please scan QR Code";
+  bool camStarted = false;
+  BarcodeCapture? currentBarcodeCapture;
+  final MobileScannerController controller = MobileScannerController(
+    facing: CameraFacing.back,
+    formats: const [BarcodeFormat.qrCode],
+    autoStart: false,
+  );
+
+  @override
+  dispose() {
+    controller.dispose();
+    super.dispose();
+  }
+
+  void startCamera() {
+    setState(() {
+      camStarted = !camStarted;
+      controller.start();
+    });
+  }
+
+  void onBarcodeDetect(BarcodeCapture barcodeCapture) {
```

Comment:
The display value can be null. We can use the raw value as fallback.
In the case where that is also null, just use a message like this?

```suggestion
  void onBarcodeDetect(BarcodeCapture barcodeCapture) {
    final barcode = barcodeCapture.barcodes.last;
  
    setState(() {
      overlayText = barcode.displayValue ?? barcode.rawValue ?? 'Barcode has no displayable value';
    });
  }
```